FRAuth/FRAuth/Authentication/NodeNext.swift

```diff
@@ -0,0 +1,224 @@
+// 
+//  NextProtocol.swift
+//  FRAuth
+//
+//  Copyright (c) 2024 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+import FRCore
+
+public class NodeNext: NSObject {
+    
+    /// A list of Callback for the state
+    @objc public var callbacks: [Callback] = []
+    /// authId for the authentication flow
+    @objc public var authId: String?
+    /// Unique UUID String value of initiated AuthService flow
+    @objc public var authServiceId: String?
+    /// Designated AuthService name defined in OpenAM
+    var serviceName: String?
+    /// authIndexType value in AM
+    var authIndexType: String?
+    /// ServerConfig information for AuthService/Node API communication
+    var serverConfig: ServerConfig?
+    /// OAuth2Client information for AuthService/Node API communication
+    var oAuth2Config: OAuth2Client?
+    /// TokenManager instance to manage, and persist authenticated session
+    var tokenManager: TokenManager?
+    /// KeychainManager instance to persist, and retrieve credentials from storage
+    var keychainManager: KeychainManager?
+    
+    public func next<T>(completion:@escaping NodeCompletion<T>) {
+
+        if T.self as AnyObject? === Token.self {
+            next { (token: Token?, node, error) in
+                completion(token as? T, node, error)
+            }
+        }
+        else if T.self as AnyObject? === AccessToken.self {
+            next { (token: AccessToken?, node, error) in
+                completion(token as? T, node, error)
+            }
+        }
+        else if T.self as AnyObject? === FRUser.self {
+            next { (user: FRUser?, node, error) in
+                completion(user as? T, node, error)
+            }
+        }
+        else {
+            completion(nil, nil, AuthError.invalidGenericType)
+        }
+    }
+    
+    func next(completion:@escaping NodeCompletion<FRUser>) {
+        FRLog.v("Called")
+        if let currentUser = FRUser.currentUser {
```

Comment:
In AuthService before the change, this line was:
`if let currentUser = FRUser.currentUser, currentUser.token != nil {`
Are we not losing anything by not checking for currentUser.token against nil?